src/librustc/traits/select.rs

```diff
@@ -3501,21 +3516,36 @@ impl<'cx, 'tcx> SelectionContext<'cx, 'tcx> {
                 // with what our behavior should be there. -nikomatsakis
                 let InferOk { obligations, .. } = self.infcx
                     .at(&obligation.cause, obligation.param_env)
-                    .eq(target, source_trait) // FIXME -- see below
+                    .eq(target, source_ty) // FIXME -- see above
                     .map_err(|_| Unimplemented)?;
                 nested.extend(obligations);
+                */
 
-                // Register one obligation for 'a: 'b.
-                let cause = ObligationCause::new(
-                    obligation.cause.span,
-                    obligation.cause.body_id,
-                    ObjectCastObligation(target),
+                // Register obligations for `dyn TraitA1 [TraitA2...]: TraitB1 [TraitB2...]`.
```

Comment:
> The second is because otherwise encounter ambiguity errors during bootstrap, owing to the rustc_datastructures::sync::Send trait (which is not the same as the standard Send trait). 

Can't we rename the `rustc_datastructures` trait?

Reply:
@Centril Not against that, though if we do, a separate PR again makes sense. :-)

Reply:
@alexreg Sure; but ideally we file an issue cause we'll forget otherwise. :)

Reply:
Sure.

Reply:
Filed as https://github.com/rust-lang/rust/issues/66609.

Reply:
Thanks!

Reply:
So there seems to be some confusion here... @nikomatsakis is the problem the name of the trait? E.g. if we rename `rustc_data_structures::Send` to something else then we don't have bootstrap problems?

cc @Mark-Simulacrum

Reply:
I'm pretty sure it's not a matter of the name of the trait, but I could be wrong.